Exporter: Dummy UI flow for export process

jordwest opened the pull request:
~~Awaiting #500, #1670~~
Blocks #1359 

This PR implements the user interface functionality for the complete export process. 

The original designs showed a popup notification upon completion, but this was decided against in favor of non-modal notices. ~~The content of the popup has instead been moved to a `CompactCard` displayed below the export settings.~~ The completion message now appears in a global notice with a button for download.

This non-modal notification will allow a user to navigate away from the export page and still receive a notice once the download file is ready.

 - [x] Switch to using `FoldableCard` component instead of `CompactCard`
 - [x] Use global notifications instead of custom component for displaying export success/failure
 - [x] Rebase and update for Redux refactor #751 
 - [x] Update the heading and subtitle text styles
 - [x] Override chevron on the `FoldableCard` expand button with a gear icon (Awaiting #1670)
 - [x] Switch to using radio buttons for Posts, Pages, Feedback selection (p195om-2Mv-p2)
 - [x] Change the wording of the advanced settings blurb to reflect the change from checkboxes -> radio buttons

# Screenshots
<img width="587" alt="screen shot 2015-12-16 at 9 16 12 pm" src="https://cloud.githubusercontent.com/assets/416133/11852845/abd28d42-a43a-11e5-8f04-7c33f12cb591.png">

<img width="612" alt="screen shot 2015-12-16 at 9 16 32 pm" src="https://cloud.githubusercontent.com/assets/416133/11852904/f6212ee4-a43a-11e5-8174-f89b915e98bf.png">

## Export in Progress
### Basic Settings
<img width="575" alt="screen shot 2015-12-16 at 9 16 25 pm" src="https://cloud.githubusercontent.com/assets/416133/11852851/b0242c2a-a43a-11e5-92e9-604295fae37f.png">

### Advanced Settings
<img width="617" alt="screen shot 2015-12-16 at 9 16 37 pm" src="https://cloud.githubusercontent.com/assets/416133/11852900/f19d01fe-a43a-11e5-82af-3012b5b69112.png">


## Successful Export
<img width="746" alt="screen shot 2015-12-16 at 9 18 07 pm" src="https://cloud.githubusercontent.com/assets/416133/11852907/fb63fbca-a43a-11e5-81c3-29ec2cb81c59.png">

## Failed Export
<img width="751" alt="screen shot 2015-12-16 at 9 18 35 pm" src="https://cloud.githubusercontent.com/assets/416133/11852911/ff1dc584-a43a-11e5-883d-0faecb2c1fc9.png">

jordwest commented:
Would love to hear any feedback on the design @mrheston @rickybanister 
cc @dllh

dllh commented:
One thing we'll need to work out is how to make the download link disappear after some period, since the export file goes away eventually. If we're just going to poll for existence of the link, then the export process could probably just fire a job to remove the data we're polling for after some period. Probably we'll also want to remove that data upon initiation of a new export too, unless this is going to be something we can use to show multiple recent export files.

I wonder if we should also make it clear that the export file will be emailed as well, or that it's ok to nav away and back to get the export file later.

jordwest commented:
@dllh: Good point. Currently the link will disappear if the page is reloaded, so expired links will only be an issue if the page isn't reloaded within that timeframe. Although that's a valid concern especially for the desktop app that might go weeks between refreshes. Clicking the export button again currently removes the export information.

dllh commented:
Seems like something we can raise an issue for and not block this over. I think the same is true of my comment about setting expectations re email or navigating away.

From my perspective, the design here is great, but we should probably get a thumbs up from an actual designer before moving forward. :)

rickybanister commented:
I discussed with @jordwest @dllh and got some feedback from @folletto today and we settled on using the FoldableCard component to refactor the Export interface, and putting the success and error messages in notices at the top of the page. 

This screenshot shows all four states at the same time:

<img width="724" alt="screen shot 2015-12-04 at 11 44 21 am" src="https://cloud.githubusercontent.com/assets/437258/11595324/cceb482e-9a7c-11e5-91c1-3d0b3c924e9a.png">

rickybanister commented:
There appears to be something off about the styling of the foldable card header. It seems you're using `h1` and `h2` for the heading and subheading.

I believe the foldable card example simply puts the main text in a `div` and then uses a `small` tag for the subtext. 

Not sure if it was meant to accommodate headings.

cc @folletto

rickybanister commented:
also it would be nice if we could swap the arrow for the `gridicon-cog`

rickybanister commented:
Actually, sorry for the confusing comment regarding h1 and h2—

You can style those to be whatever you like, the component doesn't make any hard suggestions about it.

In my design I used these for the header (regardless of the wrapping tag)

`font-size: 24px;`
`font-weight: 300;`

and 

`font-size: 11px;`

for the sub heading.

folletto commented:
> You can style those to be whatever you like, the component doesn't make any hard suggestions about it.

Yeah. The component is neutral. It doesn't do any special styling in any way.  :)
Each implementation has to apply the styles as needed.

> also it would be nice if we could swap the arrow for the gridicon-cog

+1

rickybanister commented:
I noticed two small design things here:

1. why does the principal `Export` button go away when you click the cog to open the advanced settings? Since you can start many exports ad hoc, that button does not need to go away. A user might simply want to click the cog out of curiosity and then choose a full export—we shouldn't require them to click the cog a second time for that button to re-appear.

2. The sub-heading beneath 'Feedback' that explains what is included should be moved to the right to align with the 'Feedback' heading just as the form fields align with the respective labels.

This is looking really great though!

jordwest commented:
@rickybanister, @folletto: I've just pushed a few changes to this PR in response to your feedback:
 - The chevron expand button is now a cog icon (pending #1670)
 - Adjusted the styles for the header and subtitle
 - Switched to radio buttons instead of checkboxes for selecting post type (ref p195om-2Mv-p2)

The screenshots in the description have been updated.

jordwest commented:
@rickybanister: One thing I was thinking about since we switched to radio buttons is: After selecting one of the option 'Posts', 'Pages', or 'Feedback', how would you then go back to 'All' post types?

jordwest commented:
Maybe the 'Export' button could be changed to say 'Export Complete Site', so that it's clear that the button won't take into account the settings below?

rickybanister commented:
How about `Export All`

dllh commented:
I played with this a bit and it looked good to me, but it'd be good to get other eyes on the code.

dllh commented on client/state/site-settings/exporter/actions.js:
```diff
@@ -1,30 +1,84 @@
+
+export function completeExport( downloadURL ) {
+	notices.success(
+		i18n.translate( 'Your export was successful! A download link has also been sent to your email' ),
```
Could use a period at the end of the sentence here.

jordwest replied:
Fixed

dllh commented:
@omarjackman please review this PR and let's try to get it landed. The code looks good to me, but I've spent a lot less time writing/reading Calypso code than I feel like is necessary for a thumbs up.

omarjackman commented on client/state/site-settings/exporter/actions.js:
```diff
@@ -1,30 +1,84 @@
+			// This will be replaced with polling to check when the export completes
+			setTimeout( () => {
+				dispatch( completeExport( '#', 'testing-2015-01-01.xml' ) );
+				//dispatch( failExport( 'The reason for failure would be displayed here' ) );
```
This would probably be a good place for a `// TODO:` comment instead of the commented code.

jordwest replied:
This is just here for convenience - since the API calls are not connected yet, uncommenting this line and commenting out the one above it will demonstrate what happens when an export fails.
This will be removed in #1359

omarjackman commented:
@jordwest The code here looks solid to me however I haven't done much work with this new store logic to give a good review of it. Its a :+1: from me however I'd like to see someone else take a look at that portion.

dllh commented:
Chatted with @jordwest a bit in Slack, and the store logic is going to change a fair bit in #1359. I think we should land this and request more in-depth review of the modified store logic from somebody more familiar with redux once #1359 is rebased. :ship:
